src/main/scala/spray/json/package.scala

```diff
@@ -32,7 +32,7 @@ package object json {
 
 package json {
 
-  class DeserializationException(msg: String, cause: Throwable = null) extends RuntimeException(msg, cause)
+  case class DeserializationException(msg: String, cause: Throwable = null, fieldName: Option[String] = None) extends RuntimeException(msg, cause)
```

Comment:
interesting that you want to have the `fieldNames` in the exception. I find that having the type that is being serialised is more general. Perhaps have both? I resort to including this info in a string, obviously better to have exception level support https://github.com/fommil/ensime-server/blob/jerk/spray-json-shapeless/src/main/scala/org/ensime/json/package.scala#L6